style(urls): Updated link style to round chips

shobhitexe opened the pull request:
## Closes #938 

- Changed Link style to rounded chips as per figma design file 



https://github.com/twentyhq/twenty/assets/29349136/985eabcf-87c4-4744-9a98-5dc1515e2f42

charlesBochet reviewed (commented):
Thank you

RawLink is used to display phone too, so let's keep it as it is.
Could you introduce a new component: RoundedLink in the @/ui/link/components folder.
This RoundedLink would use our Chip component. Then use it in URLFields and URLCells

We also would like to have storybook tests on top of it (see EntityChip for example which is another component that uses Chip component behind the scene

shobhitexe commented:
> Thank you
> 
> RawLink is used to display phone too, so let's keep it as it is. Could you introduce a new component: RoundedLink in the @/ui/link/components folder. This RoundedLink would use our Chip component. Then use it in URLFields and URLCells
> 
> We also would like to have storybook tests on top of it (see EntityChip for example which is another component that uses Chip component behind the scene



> Thank you
> 
> RawLink is used to display phone too, so let's keep it as it is. Could you introduce a new component: RoundedLink in the @/ui/link/components folder. This RoundedLink would use our Chip component. Then use it in URLFields and URLCells
> 
> We also would like to have storybook tests on top of it (see EntityChip for example which is another component that uses Chip component behind the scene

Thank you for the instructions. I'll create a new component just a small doubt to keep things organized, should i open a new PR for these changes or continue in this If you have any other preferences or details, please let me know. Thanks!

charlesBochet commented:
> Thank you for the instructions. I'll create a new component just a small doubt to keep things organized, should i open a new PR for these changes or continue in this If you have any other preferences or details, please let me know. Thanks!

You can keep working on this one :)

charlesBochet commented:
@shobhitexe Any update on this one?

shobhitexe commented:
> @shobhitexe Any update on this one? 🙂

On it will push changes soon

shobhitexe commented:
Hey having small doubt 

have made RoundedChip component which is using Chip component as backbone 
![image](https://github.com/twentyhq/twenty/assets/29349136/ccb27c95-5b23-4887-8c05-d92566a19edd)


the Chip component has three variant and none of them matching the style same as figma 
![image](https://github.com/twentyhq/twenty/assets/29349136/e57af281-1747-4143-afe9-36d053487dd7)


So should i introduce new variant "rounded" to ChipVariant or should i continue with existing given variant in which using "highlighted" variant

charlesBochet commented:
Yes, please introduce a new ChipVariant to do so! Do not forget to add it to our storybook stories too (see Chip.stories.tsx)

shobhitexe commented:
Hey I've made some changes 

- Added a new "Rounded" variant for chips.
- Created a new component called RoundedLink, using Chip component as backbone.
- Implemented RoundedLink in URL cells and URL fields.
- Set up a Storybook for the RoundedLink component, while the rounded variant's Chip Storybook was auto-generated and untouched."

charlesBochet commented on front/src/modules/ui/chip/components/Chip.tsx:
```diff
@@ -38,8 +39,18 @@ const StyledContainer = styled.div<Partial<OwnProps>>`
+  border-color: ${({ theme, variant }) =>
+    variant === ChipVariant.Rounded ? theme.border.color.medium : 'none'};
+  border-radius: ${({ theme, variant }) =>
+    variant === ChipVariant.Rounded ? '50px' : theme.border.radius.sm};
```
'50px' => you can use theme.border.radius.rounded

shobhitexe replied:
![image](https://github.com/twentyhq/twenty/assets/29349136/1dc8a7c7-b12a-443e-b2ee-cbaf88a7912d)


but it will look like this after applying rounded

charlesBochet commented on front/src/modules/ui/input/url/components/URLTextInputDisplay.tsx:
```diff
@@ -1,9 +1,9 @@
+const StyledRawLink = styled(RoundedLink)`
   overflow: hidden;
 
   a {
```
this a styling is not needed! It should already be handled by RoundedLink itself

charlesBochet replied:
overflow property too

charlesBochet commented on front/src/modules/ui/link/components/RawLink.tsx:
```diff
@@ -20,11 +20,15 @@ const StyledClickable = styled.div`
 `;
 
 export function RawLink({ className, href, children, onClick }: OwnProps) {
+  console.log(children);
```
console.log!

charlesBochet commented on front/src/modules/ui/link/components/RawLink.tsx:
```diff
@@ -20,11 +20,15 @@ const StyledClickable = styled.div`
-        {children}
-      </ReactLink>
-    </StyledClickable>
+    <div>
```
this div seems redundant

charlesBochet commented on front/src/modules/ui/link/components/RoundedLink.tsx:
```diff
@@ -0,0 +1,42 @@
+  }
+`;
+
+export function RoundedLink({ children, href, onClick }: OwnProps) {
```
Let's remove children as a prop and add a label: string instead, this would make the component API clearer

charlesBochet commented on front/src/modules/ui/link/components/RoundedLink.tsx:
```diff
@@ -0,0 +1,42 @@
+export function RoundedLink({ children, href, onClick }: OwnProps) {
+  return (
+    <div>
+      {children !== '' ? (
```
then we can remove this condition

shobhitexe replied:
This condition filters out links which are empty and avoids rendering them if we remove it then something like the below SS appears on empty links

![image](https://github.com/twentyhq/twenty/assets/29349136/6b3e43fb-0bec-48e6-a625-72e883ea763f)

![image](https://github.com/twentyhq/twenty/assets/29349136/895b081c-3a23-4599-a0b9-5bcf4c8c61e8)

charlesBochet replied:
Got it thank you!

charlesBochet reviewed (commented):
Nice! A last round of comments and we are good to go!

charlesBochet reviewed (approved):
LGTM, thanks you!
